roles/build_project_repo/tasks/python_app_version.yml

```diff
@@ -0,0 +1,41 @@
+# Determine the version of the python app being deployed
+- name: Determine python app version
+  tags:
+    - python
+  block:
+    - name: Check if project has a version.py file
+      stat:
+        path: "{{ clone_root}}/{{ repo }}/{{ python_app }}/version.py"
+      register: python_version_file
+
+    - name: Get the version for the python package/app being deployed (app.version)
+      become: true
+      become_user: "{{ deploy_user }}"
+      shell:
+        cmd: "python3 -c 'from {{ python_app }} import version; print(version.__version__)'"
+        chdir: "{{ clone_root}}/{{ repo }}"
+      register: python_app_version
+      # when skipped, registers a variable with info with skip / conditional info
+      when:
+        - python_version_file.stat.exists
+
+    - name: debug python app version
+      ansible.builtin.debug:
+        msg: python_app_version = {{ python_app_version}}
+
+    - name: Get the version for the python package/app being deployed (app.__version__)
+      become: true
+      become_user: "{{ deploy_user }}"
+      shell:
+        cmd: "python3 -c 'import {{ python_app }}; print({{ python_app }}.__version__)'"
+        chdir: "{{ clone_root}}/{{ repo }}"
+      register: python_app_version
```

Comment:
This doesn't overwrite the variable with skip info if skipped?

Reply:
oh good catch, it might! I didn't have a way to test that case

If you can confirm (or fix your geotaste version and I'll check), then we can go back to something more like what you had, where we set two different variables. That's probably cleaner anyway, it felt a little weird to use set fact with the same variable name.